cellprofiler/modules/imagegradient.py

```diff
@@ -0,0 +1,102 @@
+"""
+
+Image gradient
+
+"""
+
+import cellprofiler.image
+import cellprofiler.module
+import cellprofiler.setting
+import numpy
+import skimage.exposure
+import skimage.filters
+import skimage.morphology
+
+
+class ImageGradient(cellprofiler.module.Module):
+    category = "Image Processing"
+    module_name = "ImageGradient"
+    variable_revision_number = 1
+
+    def create_settings(self):
+        self.x_name = cellprofiler.setting.ImageNameSubscriber(
+            "Input"
+        )
+
+        self.y_name = cellprofiler.setting.ImageNameProvider(
+            "Output",
+            "OutputImage"
+        )
+
+        self.structure = cellprofiler.setting.Choice(
+            "Structuing element",
+            [
+                "Ball"
+            ]
+        )
+
+        self.radius = cellprofiler.setting.Integer(
+            "Radius",
+            1
+        )
+
+    def settings(self):
+        return [
+            self.x_name,
+            self.y_name,
+            self.structure,
+            self.radius
+        ]
+
+    def visible_settings(self):
+        return [
+            self.x_name,
+            self.y_name,
+            self.structure,
+            self.radius
+        ]
+
+    def run(self, workspace):
+        x_name = self.x_name.value
+
+        y_name = self.y_name.value
+
+        radius = self.radius.value
+
+        structure = self.structure.value
+
+        images = workspace.image_set
+
+        x = images.get_image(x_name)
+
+        x_data = x.pixel_data
+
+        x_data = skimage.img_as_uint(x_data)
+
+        disk = skimage.morphology.disk(radius)
+
+        y_data = numpy.zeros_like(x_data)
+
+        for z, image in enumerate(x_data):
```

Comment:
I would expect a global gradient (i.e. use `gradient` from `skimage.filters`).

It has the added benefit of working on both two-and-three dimensional images.